Objects/frameobject.c

```diff
@@ -1091,9 +1105,20 @@ fastlocalsproxy_write_to_frame(fastlocalsproxyobject *flp, PyObject *key, PyObje
     if (fast_ref != NULL) {
         /* Key is also stored on the frame, so update that reference */
         if (PyCell_Check(fast_ref)) {
-            result = PyCell_Set(fast_ref, NULL);
+            result = PyCell_Set(fast_ref, value);
         } else {
-            /* TODO: It's Python integer mapping into the fast locals array */
+            /* Fast ref is a Python int mapping into the fast locals array */
+            Py_ssize_t offset = PyLong_AsSsize_t(fast_ref);
+            Py_ssize_t max_offset = flp->frame->f_code->co_nlocals;
+            if (offset < 0) {
+                result = -1;
+            } else if (offset > max_offset) {
+                PyErr_Format(PyExc_SystemError,
+                             "Fast locals ref (%d) exceeds array bound (%d)",
```

Comment:
`%zd` needs to be used with `Py_ssize_t` (these errors were fixed recently in d53fe5f407ff4b529628b01a1bcbf21a6aad5c3a / https://bugs.python.org/issue36254).

Reply:
Fixed